src/geo/leaflet/leaflet-cartodb-layer-group-view.js

```diff
@@ -215,27 +202,19 @@ var LeafletCartoDBLayerGroupView = L.TileLayer.extend({
   },
 
   /**
-   * Update CartoDB layer
-   * generates a new url for tiles and refresh leaflet layer
+   * On tileJSON change,
+   * it generates a new url for tiles and refresh leaflet layer
    * do not collide with leaflet _update
    */
-  __update: function (done) {
-    var self = this;
-    this.fire('updated');
-    this.fire('loading');
-    var map = this.options.map;
-
-    var tilejson = self.model.get('urls');
+  _onTileJSONChange: function () {
+    var tilejson = this.model.get('urls');
     if (tilejson) {
-      self.tilejson = tilejson;
-      self.setUrl(self.tilejson.tiles[0]);
+      this.tilejson = tilejson;
+      this.setUrl(this.tilejson.tiles[0]);
       // manage interaction
-      self._reloadInteraction();
-      self.ok && self.ok();
```

Comment:
I've removed this because ok function only triggers an event called `tileOk`. Is it necessary? Rings the bell @javisantana?
I've checked rest of code and nobody in CartoDB.js is listening that event.
http://www.gifbay.com/gif/bola_del_desierto-143119/